lib.php

```diff
@@ -81,7 +80,51 @@ public static function has_config() {
 
     public static function get_config_options() {
 
-        // Get default config;
+        $defaultconfig = self::get_default_config();
+
+        $config = array();
+        $config['generalsettings']            = self::define_general_settings($defaultconfig);
+        $config['transfersettings']           = self::define_transfer_settings($defaultconfig);
+        $config['objectfssettingsfilesystem'] = self::define_client_selection($defaultconfig);
+        $clients = self::get_client_components('client');
+
+        foreach ($clients as $name => $classname) {
+            /** @var \module_objectfs\client\object_client $client */
+            $client = new $classname($defaultconfig);
+            $config[$name . 'settings'] = $client->define_settings_form();
+        }
+        return array(
+            'elements' => $config,
+        );
+
+    }
+
+    public static function define_client_selection($config) {
```

Comment:
Probably a dumb question, but are only allowing one external objectfs to be active at any one time? So at the moment, I've got this weird situation where I can select between \module_objectfs\s3_file_system and \module_objectfs\azure_file_system but my instance was previously configured with the s3_file_system and thus I see Amazon S3 settings. If we only allow one type, should we then disallow settings/disable the other type?